modules/extensions/src/terrain/shader-module.ts

```diff
@@ -0,0 +1,143 @@
+import type {_ShaderModule as ShaderModule} from '@deck.gl/core';
+
+import type {Texture2D} from '@luma.gl/core';
+
+export type TerrainModuleSettings = {
+  pickingActive?: boolean;
+  heightMap: Texture2D | null;
+  dummyHeightMap: Texture2D;
+  terrainCover: Texture2D | null;
+  terrainCoverBounds?: [number[], number[]] | null;
+  drawToTerrainHeightMap?: boolean;
+  useTerrainHeightMap?: boolean;
+  terrainSkipRender?: boolean;
+};
+
+/** A model can have one of the following modes */
+const TERRAIN_MODE = {
+  NONE: 0,
+  /** A terrain layer rendering encoded ground elevation into the height map */
+  WRITE_HEIGHT_MAP: 1,
+  /** An offset layer reading encoded ground elevation from the height map */
+  USE_HEIGHT_MAP: 2,
+  /** A terrain layer rendering to screen, using the cover fbo overlaid with its own texture */
+  USE_COVER: 3,
+  /** A terrain layer rendering to screen, using the cover fbo as texture */
+  USE_COVER_ONLY: 4,
+  /** Draped layer is rendered into a texture, and never to screen */
+  SKIP: 5
+};
+
+const TERRAIN_MODE_CONSTANTS = Object.keys(TERRAIN_MODE)
```

Comment:
I would actually prefer if this shader chunk was typed out rather than an iteration.

Reply:
We do this for all the projection enums: 
https://github.com/visgl/deck.gl/blob/6f8700f58acfd1b90f083092d65c8096c4ccc344/modules/core/src/shaderlib/project/project.glsl.ts#L24-L32

I do find it easier to keep the shader constants up to date with the JS enum that way.